geniza/corpus/models.py

```diff
@@ -329,7 +329,8 @@ def shelfmark(self):
         return " + ".join(
             dict.fromkeys(
                 block.fragment.shelfmark
-                for block in self.textblock_set.filter(certain=True)
+                for block in self.textblock_set.all()
+                if block.certain  # filter locally instead of in the db
```

Comment:
weird that this was faster...did it also change/speed up for e.g. the admin list view? I thought at one point I optimized the `get_queryset()` for a list view to prefetch textblocks so that this would be faster, so it'd be a shame to have lost that

Reply:
but i guess prefetching only certain ones is impossible, so it's one query either way, and i can see how making a bunch more per document might be inefficient

Reply:
you added optimization specifically for this? 

I have a separate branch where I'm working on the admin csv download speedup, and will share all my refactoring and time testing notes soon if you're interested :-) 

I did add some prefetching and select related to the queryset used for this export; full thing is downloading locally for me in a little under 2 minutes, which seems like it might be good enough? (although I wish it was faster)